pkg/templating/template_description.go

```diff
@@ -33,8 +36,98 @@ func TemplateStringRaw(inputTemplate string, issue *v1.Issue) (*string, error) {
 	return &res, nil
 }
 
+type enrichedIssue struct {
+	*v1.EnrichedIssue
+	ToolName       string
+	ScanStartTime  string
+	ScanID         string
+	ConfidenceText string
+	SeverityText   string
+	Count          int
+	FirstFound     string
+}
+type enrichedIssueOption func(*enrichedIssue) error
+
+func EnrichedIssueWithToolName(toolname string) enrichedIssueOption {
+	return func(ei *enrichedIssue) error {
+		if toolname == "" {
+			return errors.New("invalid toolname ''")
+		}
+		ei.ToolName = toolname
+		return nil
+	}
+}
+
+func EnrichedIssueWithScanStartTime(startTime *time.Time) enrichedIssueOption {
+	return func(ei *enrichedIssue) error {
+		if startTime == nil {
+			return errors.New("invalid startTime nil")
+		}
+		ei.ScanStartTime = startTime.Format(time.RFC3339)
+		return nil
+	}
+}
+
+func EnrichedIssueWithConfidenceText(confidence string) enrichedIssueOption {
+	return func(ei *enrichedIssue) error {
+		if confidence == "" {
+			return errors.New("invalid confidence ''")
+		}
+		ei.ConfidenceText = confidence
+		return nil
+	}
+}
+
+func EnrichedIssueWithSeverityText(severity string) enrichedIssueOption {
+	return func(ei *enrichedIssue) error {
+		if severity == "" {
+			return errors.New("invalid severity ''")
+		}
+		ei.SeverityText = severity
+		return nil
+	}
+}
+
+func EnrichedIssueWithCount(count int) enrichedIssueOption {
+	return func(ei *enrichedIssue) error {
+		if count <= 0 {
+			return errors.Errorf("invalid count %d", count)
+		}
+		ei.Count = count
+		return nil
+	}
+}
+
+func EnrichedIssueWithScanID(scanID string) enrichedIssueOption {
+	return func(ei *enrichedIssue) error {
+		if scanID == "" {
+			return errors.New("invalid scanID ")
+		}
+		ei.ScanID = scanID
+		return nil
+	}
+}
+
+func EnrichedIssueWithFirstFound(firstFound *time.Time) enrichedIssueOption {
+	return func(ei *enrichedIssue) error {
+		if firstFound == nil {
+			return errors.New("invalid firstFound nil")
+		}
+		ei.FirstFound = firstFound.Format(time.RFC3339)
+		return nil
+	}
+}
+
 // TemplateStringEnriched applies the provided go template to the Enriched Issue provided and returns the resulting str
-func TemplateStringEnriched(inputTemplate string, issue *v1.EnrichedIssue) (*string, error) {
+func TemplateStringEnriched(inputTemplate string, issue *v1.EnrichedIssue, opts ...enrichedIssueOption) (*string, error) {
+	enrichedIssue := &enrichedIssue{
+		EnrichedIssue: issue,
+	}
+	for _, opt := range opts {
+		if err := opt(enrichedIssue); err != nil {
+			return nil, err
```

Comment:
I'd use something like `return nil, fmt.Errorf("could not apply enriched issue option: %w", err)` to make it more obvious where the error is coming from